stac_mjx/controller.py

```diff
@@ -506,28 +510,61 @@ def render(
         render_mj_model, body_site_idxs, keypoint_site_idxs = (
             self._create_keypoint_sites()
         )
-        render_mj_model.site_pos[body_site_idxs] = offsets
+
+        # Add body sites for new offset positions
+        for (key, v), pos in zip(
+            self.cfg.model.KEYPOINT_MODEL_PAIRS.items(), offsets.reshape((-1, 3))
+        ):
+            parent = self._root.find("body", v)
+            parent.add(
+                "site",
+                name=key + "_new",
+                type="sphere",
+                size=[0.005],
+                rgba="0 0 0 1",
+                pos=pos,
+                group=2,
+            )
+
+        # tendons from new marker sites to kp
+        if show_marker_error:
+            for key, v in self.cfg.model.KEYPOINT_MODEL_PAIRS.items():
+                # pos = utils.params["KEYPOINT_INITIAL_OFFSETS"][key]
+                rgba = self.cfg.model.KEYPOINT_COLOR_PAIRS[key]
+                tendon = self._root.tendon.add(
+                    "spatial",
+                    name=key + "-" + v,
+                    width="0.001",
+                    rgba="255 0 0 1",  # Red
+                    limited=False,
+                )
+                tendon.add("site", site=key + "_kp")
+                tendon.add("site", site=key + "_new")
+
+        physics = mjcf.Physics.from_mjcf_model(self._root)
+        render_mj_model = deepcopy(physics.model.ptr)
 
         scene_option = mujoco.MjvOption()
+        scene_option.geomgroup[1] = 0
         scene_option.geomgroup[2] = 1
+
         scene_option.sitegroup[2] = 1
 
-        scene_option.sitegroup[3] = 1
+        scene_option.sitegroup[3] = 0
```

Comment:
What are these magic numbers?

Reply:
They determine whether that group of geom/sites will be rendered. I updated it to hide `sitegroup[3]` which contains the initial offset sites.